### ISSUE TITLE
Question about truncation trick in Big GAN.

### ISSUE THREAD

**Issue by ShenYujun** [ajbrock/BigGAN-PyTorch#14]
I think the PyTorch implementation slightly differs from that from Tensorflow Hub w.r.t. truncation trick. Here is the situation: when I set truncation as 1, these two models (tf model v.s. pth model converted from tf model) produce same results. However, it no longer holds when truncation is less than 1.

According to the code on `https://colab.research.google.com/github/tensorflow/hub/blob/master/examples/colab/biggan_generation_with_tf_hub.ipynb`, the truncation factor not only works on the input latent vector, but is also passed to the module by a placeholder. Differently, in this PyTorch version, truncation factor only affects the input noise `z` in `sample.py`.

I have also found that the following two settings give same outputs:
(1) With Tensorflow Hub,
```
z = 0.02 * truncnorm.rvs(-2, 2, size=(1, dim_z), random_state=np.random.RandomState(0))
truncation = 1.0
```
(2) With PyTorch,
```
z = 0.02 * truncnorm.rvs(-2, 2, size=(1, dim_z), random_state=np.random.RandomState(0))
```

But the following two settings give different syntheses:
(3) With Tensorflow Hub,
```
z = 0.02 * truncnorm.rvs(-2, 2, size=(1, dim_z), random_state=np.random.RandomState(0))
truncation = 0.02
```
(4) With PyTorch,
```
z = 0.02 * truncnorm.rvs(-2, 2, size=(1, dim_z), random_state=np.random.RandomState(0))
```

Would you please help me figure this thing out? I want to know how the official Big GAN play the truncation trick besides truncating the latent code. BTW, I experimented on the 512x512 BigGAN model (`https://tfhub.dev/deepmind/biggan-512/2`).

**Comment by ajbrock** [comment]
Hi ShenYujun,

The difference is that the TFHub version has a pre-computed set of BatchNorm statistics that it uses depending on the truncation level, as the distribution of activations will vary as the statistics of the latent vector are changed. As mentioned in the README in this repo's TFHub folder, this port does not support this particular setting, and you will need to accumulate new stats yourself in order to use truncation.
